crawler.py

```diff
@@ -102,14 +102,26 @@ def scrape_pages(self, aspect_list):
                 response.close()
                 if response.status is 200:
                     save_html(body, url)
-                    print("Finished crawling " + url)
+                    print("Finished crawling ", url)
+                else:
+                    print(response.status)
 
 
 def save_html(html, page):
-    print(page)
+
+    warning = """
+            <div style="position:fixed;    bottom:0;left:0;    border-top-right-radius: 8px;    color:  white;
+            background-color: red;  padding: .5em;">
+                This is a read-only mirror of the OSF.
+            </div>
+            """
+    sliding_footer = """<div id="footerSlideIn" style="display: block;">"""
+    no_sliding_footer = """<div id="footerSlideIn" style="display: none;">"""
+
+    html = html.decode('utf-8').replace('</body>', warning + '</body>').replace(sliding_footer, no_sliding_footer)
```

Comment:
Would there be a performance issue since the replace() method would probably do a string comparison and the html decoded seems to be quite a long string.